src/model/misc.rs

```diff
@@ -8,81 +8,178 @@ use std::error::Error as StdError;
 use std::result::Result as StdResult;
 #[cfg(all(feature = "model", feature = "utils"))]
 use std::str::FromStr;
-#[cfg(all(feature = "model", feature = "utils"))]
 use std::fmt;
 #[cfg(all(feature = "model", any(feature = "cache", feature = "utils")))]
 use crate::utils;
 
 /// Allows something - such as a channel or role - to be mentioned in a message.
 pub trait Mentionable {
-    /// Creates a mentionable string, that will be able to notify and/or create
-    /// a link to the item.
-    fn mention(&self) -> String;
+    /// Creates a [`Mention`] that will able to notify or create a link to the
+    /// item.
+    ///
+    /// [`Mention`] implements [`Display`], so [`ToString::to_string()`] can
+    /// be called on it, or inserted directly into a [`format_args!`].
```

Comment:
```suggestion
    /// be called on it, or inserted directly into [`format_args!`].
```

Reply:
This one in particular is referring to multiple macros of that family; `write!` `format!` `writeln!`, but I have no issue changing it.

Reply:
in that case instead of removing `a` it may be more appropriate  to say "[`format_args!`] type macro."